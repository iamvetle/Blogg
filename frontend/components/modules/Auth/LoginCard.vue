<template>
	<div class="p-6 space-y-4 md:space-y-6 sm:p-8 bg-surfaceContainerLowest rounded-lg text-onSurface" v-if="generalStore.isAuthenticated === false">
		<h1 class="text-xl font-bold leading-tight tracking-tight md:text-2xl">
			Sign in to your account
		</h1>
		<div id="form" class="space-y-4 md:space-y-6">
			<FormKit id="login_form" type="form" submit-label="Sign in" @submit="submitForm" :actions="false">
				<div>
					<FormKit type="text" name="username" label="Brukernavn" placeholder="jane43" validation="required"
						label-class="prose text-onSurface"
						input-class="prose border border-gray-300 sm:text-sm rounded-lg focus:ring-primary focus:border-primary block w-full p-2.5"
						message-class="prose text-onSurface text-sm text-red-500" />
				</div>

				<div class="mt-2">
					<FormKit type="password" name="password" label="Password" validation="required"
						placeholder="************" label-class="prose text-onSurface"
						input-class="prose border border-gray-300 sm:text-sm rounded-lg focus:ring-primary focus:border-primary block w-full p-2.5"
						message-class="prose text-sm text-red-500  text-onSurface" />
				</div>
				<div id="button" class="mt-4 shadow-sm">
					<FormKit type="submit" label="Bekreft"
						input-class="w-full text-onPrimary bg-primary focus:ring-4 focus:outline-none focus:ring-primary font-medium rounded-lg text-sm px-5 py-2.5 text-center" />
				</div>

				<p v-show="loginerror" class="mt-5 text-red-700">
					Invalid credentials
				</p>
				<p v-show="loginsucess" class="mt-5 text-green-700">
					Login successfull
				</p>
				<p class="text-sm  text-onSurface font-light mt-3">
					Don't have an account yet?
					<nuxt-link to="/registrer/" class="font-medium text-primary hover:underline">
						Sign up
					</nuxt-link>
				</p>
			</FormKit>
		</div>
	</div>
</template>

<script setup lang="ts">
//@ts-nocheck
import { useGeneralStore } from '~/store/generalStore';

const generalStore = useGeneralStore()
const loginerror = ref(false);
const loginsucess = ref(false);

const baseURL = "http://localhost:8888/api/login/";
const { redirect } = defineProps(["redirect"]);

const submitForm = async (formData:any) => {
	const response = await postForm(baseURL, formData)

	if (response.data?.token != null) {

		setLocalInfo(response.data.token, response.data.username);

		loginsucess.value = true;
		loginerror.value = false;

			if (redirect === true) {
				return navigateTo("minkonto");
			}
	} else {
		loginsucess.value = false;
		loginerror.value = true;
	}
}
</script>
