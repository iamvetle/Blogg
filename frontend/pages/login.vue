<template>
    <div
    v-if="generalStore?.isAuthenticated == false" class="flex flex-col items-center justify-center px-6 py-8 mx-auto lg:py-0 mt-24"
    >
      <div
        class="w-full bg-surfaceContainerLowest border-surfaceContainerLowest border rounded-lg shadow md:mt-0 sm:max-w-md xl:p-0"
      >
        <div class="p-6 space-y-4 md:space-y-6 sm:p-8">
          <div id="login_form">
            <LoginCard :redirect="true" />
          </div>
        </div>
      </div>
    </div>
</template>

<script setup lang="ts">



definePageMeta({
  layout:"form-layout"
})

const generalStore = useGeneralStore()

</script>

<style scoped></style>
