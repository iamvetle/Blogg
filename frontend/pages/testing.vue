<template>
  <div class="flex">

    <form @submit.prevent method="post" class="flex flex-col" enctype="multipart/form-data">

      <label for="name">Name:</label>
      <input type="text" name="name" id="name">

      <label for="email">Email</label>
      <input type="email" name="email" id="email">

      <label for="message">Message:</label>
      <textarea name="message" id="" cols="30" rows="10" placeholder="Write a message here"></textarea>

      <label for="file">File:</label>
      <input type="file" name="file" id="file">

      <button type="submit">Submit</button>
    </form>

  </div>
</template>

<script setup lang="ts">
onMounted(() => {

  const form = document.querySelector("form")

  form?.addEventListener("submit", (e: any) => {


    const formData = new FormData(form)

    for (let item of formData) {
      console.log(item[0], item[1])
    }
    
    // ta axios of prÃ¸v fetch her
  })





})

// watchEffect(() => {

// })

</script>

<style scoped></style>