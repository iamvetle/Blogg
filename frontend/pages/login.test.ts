import { mount } from '@vue/test-utils'
import login from './login.vue'
import LoginCard from '~/components/modules/Auth/LoginCard.vue'
import { createTestingPinia } from '@pinia/testing'

let wrapper
let pinia = createTestingPinia()
let generalStore = useGeneralStore(pinia)

vi.stubGlobal("definePageMeta", () => {
    return null
})


const factory = () => {
    return mount(login, {
        global: {
            components: {
                LoginCard
            },
            stubs: {
                "LoginCard":true
            },
            plugins: [pinia]
        }
    })
}

describe('login page testing', () => {

    test('Should NOT render loginform when the user is authenticated', async () => {

        // Arrange
        wrapper = factory()

        await (wrapper.vm as any).$nextTick()

        const login = wrapper.findComponent({ name: "LoginCard" })

        // no act?

        // Assert
        expect(wrapper.exists()).toBe(true)
        expect(login.exists()).toBe(false)
    })

    test('Should render the loginform when the user is not authenticated', async () => {
        // Arrange
        wrapper = factory()
        
        await (wrapper.vm as any).$nextTick()

        const loginCard = wrapper.findComponent({ name: "LoginCard" })

        // no act?

        // Assert        
        expect(loginCard.exists()).toBe(true)
    })
}
)
