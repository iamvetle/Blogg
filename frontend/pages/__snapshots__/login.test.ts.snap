// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`login page testing > Should match snapshot 1`] = `
"<div class=\\"flex flex-col items-center justify-center px-6 py-8 mx-auto lg:py-0 mt-24\\">
  <div class=\\"w-full bg-surfaceContainerLowest border-surfaceContainerLowest border rounded-lg shadow md:mt-0 sm:max-w-md xl:p-0\\">
    <div class=\\"p-6 space-y-4 md:space-y-6 sm:p-8\\">
      <div id=\\"login_form\\">
        <login-card-stub redirect=\\"true\\"></login-card-stub>
      </div>
    </div>
  </div>
</div>"
`;
